Build failure with mtl-2.3

```
Building library for xeno-0.5..
[1 of 7] Compiling Control.Spork    ( src/Control/Spork.hs, dist/build/Control/Spork.o, dist/build/Control/Spork.dyn_o )
[2 of 7] Compiling Xeno.DOM.Internal ( src/Xeno/DOM/Internal.hs, dist/build/Xeno/DOM/Internal.o, dist/build/Xeno/DOM/Internal.dyn_o )
[3 of 7] Compiling Xeno.Types       ( src/Xeno/Types.hs, dist/build/Xeno/Types.o, dist/build/Xeno/Types.dyn_o )

src/Xeno/Types.hs:18:1: warning: [-Wunused-imports]
    The import of ‘Control.Monad.Fail’ is redundant
      except perhaps to import instances from ‘Control.Monad.Fail’
    To import instances alone, use: import Control.Monad.Fail()
   |
18 | import Control.Monad.Fail
   | ^^^^^^^^^^^^^^^^^^^^^^^^^
[4 of 7] Compiling Xeno.SAX         ( src/Xeno/SAX.hs, dist/build/Xeno/SAX.o, dist/build/Xeno/SAX.dyn_o )

src/Xeno/SAX.hs:217:20: error:
    Variable not in scope: unless :: Bool -> m () -> m b
    |
217 |         Nothing -> unless (S.null text) (textF text)
    |                    ^^^^^^

src/Xeno/SAX.hs:220:11: error:
    Variable not in scope: unless :: Bool -> m () -> m a0
    |
220 |           unless (S.null text) (textF text)
    |           ^^^^^^
```

--- a/CHANGELOG.markdown
+++ b/CHANGELOG.markdown
@@ -1,3 +1,8 @@
+    0.6
+    * GHC 9.4 compatibility :
+        * get rid of MonadFail (ST s) instance use
+    * mtl-2.3 compatibility
+
     0.4.3
     * Ensure we don't grow with a negative size in DOM parser (#48)
     * Flatten code nesting in process function (#45)

--- a/src/Xeno/SAX.hs
+++ b/src/Xeno/SAX.hs
@@ -5,9 +5,7 @@
 {-# LANGUAGE NamedFieldPuns      #-}
 {-# LANGUAGE OverloadedStrings   #-}
 {-# LANGUAGE ScopedTypeVariables #-}
-
 -- | SAX parser and API for XML.
-
 module Xeno.SAX
   ( process
   , Process(..)

--- a/src/Xeno/SAX.hs
+++ b/src/Xeno/SAX.hs
@@ -5,7 +5,7 @@
 {-# LANGUAGE NamedFieldPuns      #-}
 {-# LANGUAGE OverloadedStrings   #-}
 {-# LANGUAGE ScopedTypeVariables #-}
-{-# options_ghc -Wno-unused-imports #-}
+
 -- | SAX parser and API for XML.
 
 module Xeno.SAX
@@ -19,20 +19,21 @@ module Xeno.SAX
   , skipDoctype
   ) where
 
-import           Control.Exception
-import           Control.Monad.ST
-import           Control.Monad.State.Strict
-import           Control.Spork
-import           Data.Bits
+import           Control.Exception (throw)
+import           Control.Monad (unless)
+import           Control.Monad.ST (ST, runST)
+import           Control.Monad.State.Strict (State, evalStateT, execState, modify', lift, get, put)
+import           Control.Spork (spork)
+import           Data.Bits (testBit)
 import           Data.ByteString (ByteString)
 import qualified Data.ByteString as S
 import qualified Data.ByteString.Char8 as S8
 import qualified Data.ByteString.Unsafe as SU
-import           Data.Char(isSpace)
-import           Data.Functor.Identity
-import           Data.Semigroup
-import           Data.STRef
-import           Data.Word
+import           Data.Char (isSpace)
+import           Data.Functor.Identity (Identity(..))
+import           Data.Semigroup ()
+import           Data.STRef (newSTRef, modifySTRef', readSTRef)
+import           Data.Word (Word8, Word64)
 import           Xeno.Types
 
 

--- a/xeno.cabal
+++ b/xeno.cabal
@@ -1,5 +1,5 @@
 name: xeno
-version: 0.5
+version: 0.6
 synopsis: A fast event-based XML parser in pure Haskell
 description: A fast, low-memory use, event-based XML parser in pure Haskell.  
 build-type: Simple
@@ -10,7 +10,7 @@ license: BSD3
 license-file: LICENSE
 author: Christopher Done
 maintainer: Marco Zocca (ocramz fripost org)
-tested-with:         GHC == 8.0.1, GHC == 8.2.2, GHC == 8.4.2, GHC == 8.4.4, GHC == 8.6.5, GHC == 9.0.1
+tested-with:         GHC == 8.0.1, GHC == 8.2.2, GHC == 8.4.2, GHC == 8.4.4, GHC == 8.6.5, GHC == 9.0.1, GHC == 9.0.4
 extra-source-files:  README.md
                      CHANGELOG.markdown
                      CONTRIBUTORS.md

